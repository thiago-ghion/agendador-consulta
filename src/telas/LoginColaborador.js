import React, { useState } from "react";
import Form from "react-bootstrap/Form";
import { Button } from "react-bootstrap";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import Container from "react-bootstrap/Container";
import FloatingLabel from "react-bootstrap/FloatingLabel";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

import "../App.css";
import { logarColaborador } from "../features/loginSlice";
import { setErro } from "../features/mensagemSlice";

function LoginColaborador() {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  let usuarioInput = {};
  let senhaInput = {};

  let [usuario, setUsuario] = useState("");
  let [senha, setSenha] = useState("");

  const isFormularioValido = () => {
    if (usuario === "") {
      dispatch(setErro("Preencha o usuário do colaborador"));
      usuarioInput.focus();
      return false;
    }

    if (senha === "") {
      dispatch(setErro("Preencha a senha do colaborador"));
      senhaInput.focus();
      return false;
    }

    return true;
  };

  const logar = async () => {
    if (!isFormularioValido()) {
      return;
    }

    const resposta = await dispatch(
      logarColaborador({ usuario: usuario, senha: senha })
    );

    if (resposta.error === undefined) {
      navigate("/menuColaborador");
    } else {
      console.log("resposta", resposta);
      if (resposta.payload.senhaResetada !== undefined) {
        console.log("vai mudar a rota");
        navigate("/trocaSenhaColaborador");
      }
    }
  };

  return (
    <Container>
      <Row>
        <Col>
          <div style={{ height: "100px" }}></div>
        </Col>
      </Row>
      <Row className="justify-content-md-center">
        <Col md={4}>
          <center>
            <h2>Login</h2>
          </center>
        </Col>
      </Row>
      <Row>
        <Col>
          <div style={{ height: "20px" }}></div>
        </Col>
      </Row>
      <Form>
        <Row className="justify-content-md-center">
          <Col md={6}>
            <Form.Group className="mb-3" controlId="formUsuario">
              <FloatingLabel label="Usuário" className="mb-3">
                <Form.Control
                  type="text"
                  placeholder="Informe o usuário"
                  value={usuario}
                  autoFocus
                  onChange={(event) => {
                    setUsuario(event.target.value);
                  }}
                  ref={(input) => {
                    usuarioInput = input;
                  }}
                />
              </FloatingLabel>
            </Form.Group>
          </Col>
        </Row>
        <Row className="justify-content-md-center">
          <Col md={6}>
            <Form.Group className="mb-3" controlId="formUsuario">
              <FloatingLabel label="Senha" className="mb-3">
                <Form.Control
                  type="password"
                  placeholder="Informe a senha"
                  value={senha}
                  onChange={(event) => {
                    setSenha(event.target.value);
                  }}
                  ref={(input) => {
                    senhaInput = input;
                  }}
                />
              </FloatingLabel>
            </Form.Group>
          </Col>
        </Row>

        <Row className="justify-content-md-center">
          <Col md={6} className="align-self-end">
            <Form.Group className="mb-3" controlId="formBotaoLogin">
              <a href="/trocaSenhaColaborador" className="link-primary">
                Esqueci a senha
              </a>
            </Form.Group>
          </Col>
        </Row>

        <Row className="justify-content-md-center">
          <Col md={6}>
            <Form.Group className="mb-3" controlId="formBotaoLogin">
              <div className="d-grid gap-2">
                <Button size="lg" onClick={logar}>
                  Login
                </Button>
              </div>
            </Form.Group>
          </Col>
        </Row>
      </Form>
    </Container>
  );
}

export default LoginColaborador;
