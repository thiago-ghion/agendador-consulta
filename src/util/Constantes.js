let URL_API;

if (process.env.NODE_ENV === "development" || process.env.NODE_ENV === "test") {
  URL_API = "http://localhost:5000";
} else {
  URL_API = "https://api-agendador-consulta.herokuapp.com";
}

export { URL_API };
