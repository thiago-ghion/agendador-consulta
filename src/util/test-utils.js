import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
// As a basic setup, import your same slice reducers
import loginReducer from "../features/loginSlice";
import profissionalReducer from "../features/profissionalSlice";
import consultaReducer from "../features/consultaSlice";
import pacienteReducer from "../features/pacienteSlice";
import horarioReducer from "../features/horarioSlice";
import usuarioReducer from "../features/usuarioSlice";
import loadingReducer from "../features/loadingSlice";
import mensagemgReducer from "../features/mensagemSlice";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({
      reducer: {
        login: loginReducer,
        consulta: consultaReducer,
        profissional: profissionalReducer,
        paciente: pacienteReducer,
        horario: horarioReducer,
        usuario: usuarioReducer,
        loading: loadingReducer,
        mensagem: mensagemgReducer,
      },
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
