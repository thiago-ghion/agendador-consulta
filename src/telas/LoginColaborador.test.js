import React from "react";
import { rest } from "msw";
import { setupServer } from "msw/node";
import { renderWithProviders } from "../util/test-utils";
import { BrowserRouter as Router } from "react-router-dom";

import LoginColaborador from "./LoginColaborador";
import { fireEvent, screen, waitFor } from "@testing-library/react";

let mockLogin = {};
let estadoLogin = 200;

export const handlers = [
  rest.post(
    "http://localhost:5000/v1/seguranca/loginColaborador",
    (req, res, ctx) => {
      return res(ctx.status(estadoLogin), ctx.json(mockLogin));
    }
  ),
];

const server = setupServer(...handlers);

// Enable API mocking before tests.
beforeAll(() => server.listen());

// Reset any runtime request handlers we may add during the tests.
afterEach(() => server.resetHandlers());

// Disable API mocking after the tests are done.
afterAll(() => server.close());

beforeEach(() => {
  mockLogin = {};
  estadoLogin = 200;
});

test("Renderização componente", async () => {
  expect(
    renderWithProviders(
      <Router>
        <LoginColaborador />
      </Router>
    )
  ).toMatchSnapshot();
});

test("Usuário não preenchido ", async () => {
  renderWithProviders(
    <Router>
      <LoginColaborador />
    </Router>
  );

  const login = await screen.findByRole("button", { name: /Login/i });
  fireEvent.click(login);
});

test("Senha não preenchida", async () => {
  renderWithProviders(
    <Router>
      <LoginColaborador />
    </Router>
  );

  const usuario = await screen.findByPlaceholderText("Informe o usuário");
  fireEvent.change(usuario, { target: { value: "usuarioteste" } });
  const login = await screen.findByRole("button", { name: /Login/i });
  fireEvent.click(login);
});

test("Login com falha", async () => {
  estadoLogin = 400;

  renderWithProviders(
    <Router>
      <LoginColaborador />
    </Router>
  );

  const usuario = await screen.findByPlaceholderText("Informe o usuário");
  fireEvent.change(usuario, { target: { value: "usuarioteste" } });

  const senha = await screen.findByPlaceholderText("Informe a senha");
  fireEvent.change(senha, { target: { value: "12345678" } });

  const login = await screen.findByRole("button", { name: /Login/i });
  fireEvent.click(login);
});

test("Login com falha - senha resetada", async () => {
  estadoLogin = 400;
  mockLogin = { senhaResetada: true };

  renderWithProviders(
    <Router>
      <LoginColaborador />
    </Router>
  );

  const usuario = await screen.findByPlaceholderText("Informe o usuário");
  fireEvent.change(usuario, { target: { value: "usuarioteste" } });

  const senha = await screen.findByPlaceholderText("Informe a senha");
  fireEvent.change(senha, { target: { value: "12345678" } });

  const login = await screen.findByRole("button", { name: /Login/i });
  fireEvent.click(login);
});

test("Login com sucesso", async () => {
  renderWithProviders(
    <Router>
      <LoginColaborador />
    </Router>
  );

  const usuario = await screen.findByPlaceholderText("Informe o usuário");
  fireEvent.change(usuario, { target: { value: "usuarioteste" } });

  const senha = await screen.findByPlaceholderText("Informe a senha");
  fireEvent.change(senha, { target: { value: "12345678" } });

  const login = await screen.findByRole("button", { name: /Login/i });
  fireEvent.click(login);
});
