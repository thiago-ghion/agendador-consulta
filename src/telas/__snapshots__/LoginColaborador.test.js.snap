// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderização componente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              style="height: 100px;"
            />
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-4"
          >
            <center>
              <h2>
                Login
              </h2>
            </center>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              style="height: 20px;"
            />
          </div>
        </div>
        <form
          class=""
        >
          <div
            class="justify-content-md-center row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="mb-3"
              >
                <div
                  class="mb-3 form-floating"
                >
                  <input
                    class="form-control"
                    maxlength="15"
                    placeholder="Informe o usuário"
                    type="text"
                    value=""
                  />
                  <label>
                    Usuário
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="justify-content-md-center row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="mb-3"
              >
                <div
                  class="mb-3 form-floating"
                >
                  <input
                    class="form-control"
                    maxlength="8"
                    placeholder="Informe a senha"
                    type="password"
                    value=""
                  />
                  <label>
                    Senha
                  </label>
                </div>
              </div>
            </div>
          </div>
          <div
            class="justify-content-md-center row"
          >
            <div
              class="align-self-end col-md-6"
            >
              <div
                class="mb-3"
              >
                <a
                  class="link-primary"
                  href="/trocaSenhaColaborador"
                >
                  Trocar a senha
                </a>
              </div>
            </div>
          </div>
          <div
            class="justify-content-md-center row"
          >
            <div
              class="col-md-6"
            >
              <div
                class="mb-3"
              >
                <div
                  class="d-grid gap-2"
                >
                  <button
                    class="btn btn-primary btn-lg"
                    type="button"
                  >
                    Login
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            style="height: 100px;"
          />
        </div>
      </div>
      <div
        class="justify-content-md-center row"
      >
        <div
          class="col-md-4"
        >
          <center>
            <h2>
              Login
            </h2>
          </center>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div
            style="height: 20px;"
          />
        </div>
      </div>
      <form
        class=""
      >
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="mb-3"
            >
              <div
                class="mb-3 form-floating"
              >
                <input
                  class="form-control"
                  maxlength="15"
                  placeholder="Informe o usuário"
                  type="text"
                  value=""
                />
                <label>
                  Usuário
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="mb-3"
            >
              <div
                class="mb-3 form-floating"
              >
                <input
                  class="form-control"
                  maxlength="8"
                  placeholder="Informe a senha"
                  type="password"
                  value=""
                />
                <label>
                  Senha
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="align-self-end col-md-6"
          >
            <div
              class="mb-3"
            >
              <a
                class="link-primary"
                href="/trocaSenhaColaborador"
              >
                Trocar a senha
              </a>
            </div>
          </div>
        </div>
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-6"
          >
            <div
              class="mb-3"
            >
              <div
                class="d-grid gap-2"
              >
                <button
                  class="btn btn-primary btn-lg"
                  type="button"
                >
                  Login
                </button>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
